gateway-js/src/index.ts

```diff
@@ -545,10 +619,10 @@ export class ApolloGateway implements GraphQLService {
     }
   }
 
-  protected serviceListFromCsdl() {
+  protected serviceListFromCsdl(csdl?: DocumentNode) {
```

Comment:
Why optional? Can we just change the caller to always pass it in?

(I'll admit I'm unsure why there are protected methods in this file. Do end users subclass ApolloGateway? Do we need to keep the interfaces of protected methods consistent?)

Reply:
fb0f231